protostar/profiler/transaction_profiler_test.py

```diff
@@ -0,0 +1,223 @@
+                            Instruction(100, 0, replace(funct), 0),
+                        ],
+                        [],
+                        [],
+                        [],
+                    ),
+                ),
+                ContractProfile(
+                    ["c"],
+                    ep,
+                    RuntimeProfile(
+                        [],
+                        [
+                            Instruction(10, 0, replace(funct), 0),
+                            Instruction(100, 0, replace(funct), 0),
+                        ],
+                        [],
+                        [],
+                        [],
+                    ),
+                ),
+            ],
+            {
+                "a": 0,
+                "b": 1001,
+                "c": 1152,
+            },
+        ),
+        ([], {}),
+    ],
+)
+def test_get_instructions_id_offsets(samples, result):
+    assert get_instruction_id_offsets(samples) == result
+
+
+glob_func = {
+    "a.A": Function("A", "z", 0),
+    "a.B": Function("B", "z", 0),
+    "b.C": Function("C", "z", 0),
+}
+
+ep = ContractEntryPoint(0, 0)
+
+
+@pytest.mark.parametrize(
+    "samples,result",
+    [
+        (
+            [
+                ContractProfile(
+                    ["a"],
+                    ep,
+                    RuntimeProfile(
+                        [], [Instruction(1000, 0, glob_func["a.A"], 0)], [], [], []
+                    ),
+                ),
+                ContractProfile(
+                    ["b"],
+                    ep,
+                    RuntimeProfile(
+                        [],
+                        [
+                            Instruction(150, 0, glob_func["b.C"], 0),
+                            Instruction(100, 0, glob_func["b.C"], 0),
+                        ],
+                        [],
+                        [],
+                        [],
+                    ),
+                ),
+                ContractProfile(
+                    ["a"],
+                    ep,
+                    RuntimeProfile(
+                        [],
+                        [
+                            Instruction(1000, 0, glob_func["a.A"], 0),
+                            Instruction(100, 0, glob_func["a.B"], 0),
+                        ],
+                        [],
+                        [],
+                        [],
+                    ),
+                ),
+            ],
+            {
+                ("a.A", 1000): Instruction(1000, 0, glob_func["a.A"], 0),
+                ("b.C", 150): Instruction(1151, 0, glob_func["b.C"], 0),
+                ("b.C", 100): Instruction(1101, 0, glob_func["b.C"], 0),
+                ("a.B", 100): Instruction(100, 0, glob_func["a.B"], 0),
+            },
+        ),
+        ([], {}),
+    ],
+)
+def test_build_global_instructions(samples, result):
+    assert build_global_instructions(glob_func, samples) == result
```

Comment:
Missing types. Tests can be treated as a documentation. I start reviewing from here, and I don't know what a Sample is. Is this term defined? If so, why where is defined?

Maybe samples defined here should be moved to the conftest file and given a name.

Reply:
>Maybe samples defined here should be moved to the conftest file and given a name.

Those are used only in this file, I don't think it is a good idea